packages/taquito/test/contract/rpc-contract-provider.spec.ts

```diff
@@ -1664,6 +1676,36 @@ describe('RpcContractProvider test', () => {
     });
   });
 
+  describe('ballot', () => {
+    it('should produce a ballot operation', async (done) => {
+      const result = await rpcContractProvider.ballot({
+        proposal: 'PtKathmankSpLLDALzWw7CGD2j2MtyveTwboEYokqUCP4a1LxMg',
+        ballot: 'yay',
+      });
+
+      expect(result.raw).toEqual({
+        opbytes: 'test',
+        opOb: {
+          branch: 'test',
+          contents: [
+            {
+              source: 'test_pub_key_hash',
+              kind: 'ballot',
+              period: undefined, // value is derived from an rpc call through a protected member function in RpcContractProvider
```

Comment:
I am a bit puzzled by this undefined value that should be 1 according to the mocked `getCurrentPeriod` value. Many properties like protocol, branch and counter are fetched in the `prepareOperation` method too, and are not undefined here.

Reply:
so I was actually meaning to see whether anyone has the experience to mock this call. The prepareOperation has an RPC call to the `getCurrentPeriod` RPC endpoint, but because it none of its properties are passed through the constructor of RpcClient it fails to find the mock value properly.

I tried using `jest.spyOn` but was having failures mocking the return value of `prepareOperation`